portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.*;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet responsible for deleting all the comments in the datastore
+ */
+@WebServlet("/delete-data")
+public class DeleteDataServlet extends HttpServlet {
+
+  // Get the datastore instance that contains all of the comments.
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  /**
+   * This POST request will delete all comments in the datastore
+   * @param request the request sent from client. Body should be empty
+   * @param response response to send back to the client. Should be empty.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) {
+    // Create a query to get all Comment entities
+    Query query = new Query("Comment");
+
+    // Use the query to get results from the datastore
+    PreparedQuery results = datastore.prepare(query);
+
+    // Iterate through all entities and delete them from datastore
+    for (Entity entity: results.asIterable()) {
+      Key entityKey = entity.getKey();
+      datastore.delete(entityKey);
```

Comment:
You could consider using the overloading https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/DatastoreService.html#delete-java.lang.Iterable- to remove the need for this loop

Reply:
Looks interesting, but to use this I would need an Iterable<Keys>. I looked at all the methods for PreparedQuery in the documentation, and it doesn't look like there's a utility method to return an iterable of just Entity Keys and not full entity objects.

Due to this, it looks like I'll need a for loop no matter what. I did however find out about the .setKeysOnly() option I can put on the Query, which should make this much more efficient!

Reply:
Same as above, `stream().map()` is an option, but don't feel pressured to use it. I'm just sharing something I think might come in handy in certain situations :)